### PR TITLE
Subset URL tests by file/javascript/mailto schemes

### PR DESCRIPTION
**Pull request by foolip**

### CODE REVIEW

**Comment by gsnedders** [comment]
for all browsers this seems to (surprisingly) increase the total number of tests:

Test | master | 83f8916
-- | -- | --
/url/ | 3769 / 3819 | 3790 / 3840

**Comment by foolip** [comment]
@gsnedders there should be some increase, as there's both harness status and "Loading data…" tests contributing to the denominator. There are 7 tests that were turned into 4 variants each, which adds 7*3 = 21. That matches the delta seen.

File: url/resources/a-element.js [hunk 1]
```diff
@@ -19,7 +19,18 @@ function runURLTests(urltests) {
...
     if (expected.base === null) continue;
 
-    test(function() {
+    function getKey(expected) {
+      if (expected.protocol) {
+        return expected.protocol.replace(":", "");
+      }
+      if (expected.failure) {
+        var key = expected.input.split(":")[0];
+        return key;
```
Comment (foolip) [inline url/resources/a-element.js:28]:
```suggestion
        return expected.input.split(":")[0];
```

File: url/url-constructor.any.js [hunk 2]
```diff
@@ -9,7 +14,18 @@ function runURLTests(urltests) {
...
     if (typeof expected === "string") continue // skip comments
 
-    test(function() {
+    function getKey(expected) {
+      if (expected.protocol) {
+        return expected.protocol.replace(":", "");
+      }
+      if (expected.failure) {
+        var key = expected.input.split(":")[0];
+        return key;
```
Comment (foolip) [inline url/url-constructor.any.js:23]:
```suggestion
        return expected.input.split(":")[0];
```

File: url/url-setters-a-area.window.js [hunk 3]
```diff
@@ -15,15 +21,16 @@ function runURLSettersTests(all_test_cases) {
       if ("comment" in test_case) {
         name += " " + test_case.comment;
       }
-      test(function() {
+      var key = test_case.href.split(":")[0];
```
Comment (foolip) [inline url/url-setters-a-area.window.js:24]: @annevk said elsewhere he'd prefer:

```suggestion
      const key = test_case.href.split(":")[0];
```

File: url/url-setters.any.js [hunk 4]
```diff
@@ -15,7 +21,8 @@ function runURLSettersTests(all_test_cases) {
       if ("comment" in test_case) {
         name += " " + test_case.comment;
       }
-      test(function() {
+      var key = test_case.href.split(":")[0];
```
Comment (foolip) [inline url/url-setters.any.js:24]:
```suggestion
      const key = test_case.href.split(":")[0];
```

**Review by annevk — APPROVED** [review]
Happy to rubberstamp this.

**Pull request merged**